evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -139,8 +141,20 @@ def _transform_parameters(self, pipeline_class, proposed_parameters):
             if 'number_features' in init_params:
                 component_parameters['number_features'] = self.number_features
             # Pass the pipeline params to the components that need them
-            for param_name, value in self._pipeline_params.items():
-                if param_name in init_params:
-                    component_parameters[param_name] = value
+            if component_class.name in self._pipeline_params:
```

Comment:
I don't think we should do this because this would overwrite the parameters proposed by the tuner with a random value. I think that defeats our automl tuner because we're no longer using the pipeline scores to inform what the next parameter should be. 

I think the problem is that the default parameters of the pipeline are not necessarily in the new constrained search space. I think we should make sure of that when IterativeAlgorithm [creates the first batch](https://github.com/alteryx/evalml/blob/main/evalml/automl/automl_algorithm/iterative_algorithm.py#L66) but leave all subsequent batches alone.

All subsequent batches will have parameters within the valid ranges because we supplied the user's parameters to make_pipeline (the design document says that if a user passes in custom pipelines, it's assumed they set the right hyperparameter ranges themselves).

Reply:
Ah, good call. I'll change it so that it only sets the component parameters if we are in the first batch, and otherwise, it should skip this step.